src/database/adapters/mongo.ts

```diff
@@ -93,6 +93,9 @@ export const create = (db: string, collection: string): BidDatabaseService => {
     bid: Bid,
     archiveCollection: string
   ): Promise<boolean> => {
+    //Default cancellation date if not present
+    if (!bid.cancelDate) bid.cancelDate = new Date().getTime();
```

Comment:
Could we default to a time in the past? Like roughly when this PR is getting merged. Otherwise the cancellation time will always be changing.

Reply:
You should remove this and do one singular call to assign the `cancelDate` to a bid before the DB call in `api.ts`, this way you can know with certainty that a bid has this object and it will always be the exact same in the DB as it is in the message queue. Without doing this, the message and the db record will have different values